nav2_safety_nodes/src/nav2_safety_node.cpp

```diff
@@ -0,0 +1,232 @@
+  zone_action_ = get_parameter("zone_action").as_double();
+  zone_priority_ = get_parameter("zone_priority").as_int();
+  zone_num_pts_ = get_parameter("zone_num_pts").as_int();
+  base_frame_ = get_parameter("base_frame").as_string();
+  tf_tolerance_ = get_parameter("tf_tolerance").as_double();
+  scan_topics_ = get_parameter("scan_topics").as_string_array();
+  speed_limit_topic_ = get_parameter("speed_limit_topic").as_string();
+
+  // If the safety_polygon has been specified, it must be in the correct format
+  if (safety_polygon_ != "" && safety_polygon_ != "[]") {
+    // Polygon parameter has been specified, polygon -> point vector(safety_zone)
+    std::vector<geometry_msgs::msg::Point> safety_zone_;
+    /// string of polygon points and returns a polygon vector
+    nav2_util::makeVectorPointsFromString(safety_polygon_, safety_zone_);
+  } else {
+    // Polygon provided but invalid, so stay with the radius
+    throw std::invalid_argument("The safety_polygon is invalid");
+  }
+
+  // Initialize transform listener and broadcaster
+  tf2_ = std::make_unique<tf2_ros::Buffer>(get_clock());
+  auto timer_interface = std::make_shared<tf2_ros::CreateTimerROS>(
+    get_node_base_interface(), get_node_timers_interface());
+  tf2_->setCreateTimerInterface(timer_interface);
+  tf2_listener_ = std::make_unique<tf2_ros::TransformListener>(*tf2_);
+
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+nav2_util::CallbackReturn
+SafetyZone::on_activate(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(logger_, "Activating");
+  // Create the publishers and subscribers
+  safety_polygon_pub_ = create_publisher<geometry_msgs::msg::PolygonStamped>(
+    "published_polygon", rclcpp::SystemDefaultsQoS());
+  // Velocity publisher
+  speed_limit_pub_ = create_publisher<nav2_msgs::msg::SpeedLimit>(
+    speed_limit_topic_, rclcpp::QoS(10));
+  // Multiple Laserscan subscribers
+  scan_subscribers_.reserve(scan_topics_.size());
+  RCLCPP_INFO(logger_, "Subscribing to scan topics");
+  for (unsigned int i = 0; i < scan_topics_.size(); i++) {
+    scan_subscribers_.push_back(
+      create_subscription<sensor_msgs::msg::LaserScan>(
+        scan_topics_[i].c_str(), rclcpp::SystemDefaultsQoS(),
+        std::bind(&SafetyZone::laserCallback, this, std::placeholders::_1)));
+  }
+  // Timer -> 10hzs
+  auto update_frequency_s = std::chrono::duration<double>(update_frequency_);
+  auto update_frequency_ms = std::chrono::duration_cast<std::chrono::milliseconds>(update_frequency_s);
+  timer_ = create_wall_timer(
+  update_frequency_ms, std::bind(&SafetyZone::timerCallback, this));
+  RCLCPP_INFO(logger_, "Subscribed to scan topics");
+  safety_polygon_pub_->on_activate();
+  speed_limit_pub_->on_activate();
+
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+nav2_util::CallbackReturn
+SafetyZone::on_deactivate(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(logger_, "Deactivating");
+  safety_polygon_pub_->on_deactivate();
```

Comment:
Reset the subscriptions too for the laser scanner

Reply:
You put it in cleanup, not in deactivate. They are created in activate, so they need to be destroyed in deactivate.